ovc/client.go

```diff
@@ -137,19 +167,98 @@ func (c *Client) Do(req *http.Request) ([]byte, error) {
 	if err != nil {
 		return nil, err
 	}
+	taskID := string(body)
 
 	c.logger.Debug("OVC call: " + req.URL.Path)
 	c.logger.Debug("OVC response status code: " + resp.Status)
-	c.logger.Debug("OVC response body: " + string(body))
+	c.logger.Debug("OVC response body: " + string(taskID))
 
 	switch {
 	case resp.StatusCode == 401:
 		return nil, ErrAuthentication
 	case resp.StatusCode > 202:
-		return body, errors.New(string(body))
+		return body, errors.New(taskID)
+	}
+
+	// remove quotes from taskID if contains any
+	taskID = strings.Replace(taskID, "\"", "", -1)
+
+	// create request to get result of an async API call by job id
+	taskJSON, err := json.Marshal(
+		struct {
+			TaskID string `json:"taskguid"`
+		}{
+			TaskID: taskID,
+		},
+	)
+	if err != nil {
+		return nil, err
 	}
 
-	return body, nil
+	var taskReq *http.Response
+	result := make([]interface{}, 0)
+	start, timeout := time.Now(), 10*time.Second
+
+	// wait for result of the async task
+	for {
+		time.Sleep(2 * time.Second)
+		reqResult, err := http.NewRequest("POST", c.ServerURL+"/system/task/get", bytes.NewBuffer(taskJSON))
+		if err != nil {
+			return nil, err
+		}
+		reqResult.Header.Set("Authorization", fmt.Sprintf("bearer %s", tokenString))
+		reqResult.Header.Set("Content-Type", "application/json")
+		taskReq, err = client.Do(reqResult)
+		if taskReq != nil {
+			defer taskReq.Body.Close()
+		}
+		if err != nil {
+			return nil, err
+		}
+		switch {
+		case taskReq.StatusCode == 400:
+			if err != nil {
+				return nil, errors.New(taskID)
+			}
+			return nil, errors.New(taskID)
+		case taskReq.StatusCode == 401:
+			return nil, ErrAuthentication
+		case taskReq.StatusCode == 404:
+			// task may have not been registered yet
+			continue
+		case taskReq.StatusCode > 202:
+			return nil, errors.New(taskID)
+		}
+		if taskReq.Body != nil {
+			resultBody, err := ioutil.ReadAll(taskReq.Body)
+			if err != nil {
+				return nil, err
+			}
+			if len(resultBody) != 0 {
+				err = json.Unmarshal(resultBody, &result)
+
+				if err != nil {
+					return resultBody, nil
+				}
+				if len(result) != 0 {
+					break
+				}
+			}
+		}
+		if now := time.Now(); now.Sub(start) > timeout {
+			return nil, fmt.Errorf("job timeout %s", taskID)
+		}
+	}
+
+	success := result[0].(bool)
```

Comment:
This will panic if this is not a boolean, not sure how a failed task response looks like though

Reply:
You can do a safe check like this:

```
var firstVar interface{}
firstVar = "this is a string"

firstString, ok := firstVar.(string)

if (!ok) {
	fmt.Printf("firstString is not a string, do something about it! Value:'%v'\n", firstString)
}
```